lib/guest_installation_and_configuration_base.pm

```diff
@@ -313,11 +313,8 @@ sub revise_guest_version_and_build {
     }
 
     if ($self->{guest_build} eq '') {
-        if ((get_var('REPO_0_TO_INSTALL', '') eq '') and ($self->{guest_version} eq lc get_required_var('VERSION'))) {
-            $self->{guest_build} = lc get_required_var('BUILD');
-        }
-        elsif ((get_var('REPO_0_TO_INSTALL', '') ne '') and ($self->{guest_version} eq lc get_required_var('TARGET_DEVELOPING_VERSION'))) {
-            $self->{guest_build} = lc get_required_var('BUILD');
+        if ((!get_var('REPO_0_TO_INSTALL') and ($self->{guest_version} eq lc get_required_var('VERSION'))) or (get_var('REPO_0_TO_INSTALL') and ($self->{guest_version} eq lc get_required_var('TARGET_DEVELOPING_VERSION')))) {
+            $self->{guest_build} = lc get_var('BUILD_ISO', get_required_var('BUILD'));
```

Comment:
Maybe you can add a comment why we use BUILD_ISO here. Something like:
`BUILD_ISO is used in some cases (e.g. SLE Micro) where BUILD variable does not represent the real image build`

Reply:
Done